RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift

```diff
@@ -472,17 +472,27 @@ extension DappInteractionFlow {
 
 		func handleAccountsProofOfOwnership(
 			_ item: State.AnyInteractionItem,
-			_ accountAuthProofs: [AccountAuthProof],
-			_ signedAuthChallenge: SignedAuthChallenge
+			_ accountAddresses: [AccountAddress],
+			_ signedAuthIntent: SignedAuthIntent
 		) -> Effect<Action> {
-			let proofs = accountAuthProofs.map {
-				WalletToDappInteractionProofOfOwnership.account(.init(
-					accountAddress: $0.account.address,
-					proof: $0.proof
-				))
+			// Build account proofs
+			let accountProofs: [WalletToDappInteractionAccountProof] = signedAuthIntent.intentSignaturesPerOwner.compactMap { item in
+				guard let accountAddress = item.owner.accountAddress else {
+					return nil
+				}
+				return .init(
+					accountAddress: accountAddress,
+					proof: .init(intentSignatureOfOwner: item)
+				)
+			}
+
+			// Verify there is a signature for each address
+			guard Set(accountAddresses) == Set(accountProofs.map(\.accountAddress)) else {
+				return dismissEffect(for: state, errorKind: .failedToSignAuthChallenge, message: nil)
 			}
 
-			let challenge = signedAuthChallenge.challenge
+			let proofs = accountProofs.map { WalletToDappInteractionProofOfOwnership.account($0) }
```

Comment:
this can be `accountProofs.map(WalletToDappInteractionProofOfOwnership.account)` ?